scripts/src/submission/submission.py

```diff
@@ -527,6 +527,6 @@ def download_index_data(repository, branch="gh_pages"):
     if r.status_code == 200:
         data = yaml.load(r.text, Loader=Loader)
     else:
-        data = {}
+        data = {"apiVersion": "v1", "entries": {}}
 
```

Comment:
One final thought here (not to rehash the previous discussion), do we want to add protections that would raise an exception in the production repo vs. non production repositories?

Reply:
I was hesitant. Because according to me the better solution would be to have some index file on the gh-pages branch of the sandbox repo so that end to end can pass. But I confess haven't taken the time to test the possible consequences of doing so (and am unsure which index file to craft exactly, though probably an empty index file set to `apiVersion": "v1", "entries": {}` would suffice.)

In the end I implemented the protection measure as it was easier to achieve.

Reply:
Hrm, two things now:

1. I lean towards not having a repository hard-coded in scripts here. I think what I'd rather see is a caller receive that context as an input, and _that input_ trigger the protection. We evaluate `$GITHUB_REPOSITORY` is not the production in various places in Actions, but we don't really see a lot of references to the repository information hard-coded to impact logic (validated with `grep`, so please cross-check me on this). Let me know what you think. I'm the fence, but I lean against having it hard-coded here, and having just something we toggle in the function signature that change sits behavior, and is determined by inputs when the script is called (e.g. CLI flags, environment variables, etc. that wrap this logic).

4. You probably don't want to raise an exception when the YAML loading fails, right? You just want to catch it, and then move on so you can return the empty index.  Something like this:

```python
...
try:
    data = yaml.load(r.text, Loader=Loader)
except yaml.YAMLError as e:
    # log e if you want to
    data = {"apiVersion": "v1", "entries": {}}
return data
```

Reply:
Whoops, I should have added this to a new review. These points apply to the changes as a whole. I've marked all other comments as resolved to clarify.

Reply:
No problem, this way I can reply in the same thread and discussion isn't cluttered by linked PRs.

I pushed a change reg. (1), is this what you mean ? I would have to basically move the check to the GitHub action that runs the upcoming `validate.py` and add a CLI Flag (if ` $GITHUB_REPOSITORY="openshift-helm-charts/charts"`, then set a flag to False).

Reg. (2), in the event the function manages to fetch an index.yaml file from the repo/branch, but that index file is not valid ... Do we really want to default to an empty index ? I'd rather fail here. It means probably that something is wrong in our repo (Helm repository index is corrupted) or that the index got corrupted during download.

Reply:
Oh, duh! Thanks :laughing: Resolving this.